### ISSUE TITLE
Bug: Datepicker - selecting a date is not announced as actionable by screenreaders

### ISSUE THREAD

**Issue by BobbyJoStefflTR** [ng-bootstrap/ng-bootstrap#3677]
### Bug description:
[WCAG SC: 4.1.2 Name, Role, Value](https://www.w3.org/TR/WCAG20/#ensure-compat-rsv)
Issue: Days on the calendar are not semantic buttons, although they are selectable with enter and space keys. Screen reader users are not alerted that this is an interactive item, and may not know how to interact with it. Additionally there is no indication to a screen reader that these items are disabled and unable to be selected.

[W3C Date Picker Dialog Example](https://www.w3.org/TR/wai-aria-practices-1.1/examples/dialog-modal/datepicker-dialog.html)

**Steps to reproduce:**
In datepicker, while using a screen reader, arrow through the dates and listen for evidence of the type of element you've landed upon.

**Expected behavior:** Buttons are announced as buttons by screen readers and announced as disabled when dates are unavailable.
**Actual behavior:** Buttons are not announced as buttons. Disabled dates are not indicated to screen readers.

**Preferred solution:**
Use `<button>` element instead of `<div>` inside of the `role="gridcell"` element
**Preferred solution HTML:**
```
<div role="gridcell">
  <button ... >
      <div class="btn-light ng-star-inserted" ngbdatepickerdayview="">11</div>
  </button>
</div>
```

Given that the div already has class on it to look like a button, it makes sense to just make it a button and expose the click function `(click)="doSelect(day);"` for selecting a day on there.
![image](https://user-images.githubusercontent.com/55457806/79358917-7ef97e00-7f07-11ea-88ae-5f441a53f99c.png)

### Link to minimally-working StackBlitz that reproduces the issue:
https://stackblitz.com/edit/angular-trwjqx 

Expected behavior:
* Use buttons for selecting dates.
* Move click function to the button instead of `<div>`.
* Allow disabling of the button.

### Versions of Angular, ng-bootstrap and Bootstrap:
Angular: 9.0.7
ng-bootstrap: 6.0.2
Bootstrap: latest

**Comment by benouat** [comment]
Hey @BobbyJoStefflTR !

Thanks to report this issue. Your description is super nice and very detailed! You definitely
sounds more confortable with the subject than us! 😉 

Here is a proposal → Would you like, and most of all, would you feel ok by giving it a try via a PR ?

**Comment by maxokorokov** [comment]
If I understand correctly the proposal is to replace this `div` by a `button`, but only for the day that is selected I guess:

https://github.com/ng-bootstrap/ng-bootstrap/blob/f3a7c7d1ffc032952b62eeef4e48f0aa5765c9cf/src/datepicker/datepicker.ts#L129-L135

Just one comment about the click handling. I'm not sure we'll be able to move it on the `button` element, because the whole contents of the `<div class="ngb-dp-day">...</div>` can be replaced completely by the `[dayTemplate]` input, but we still need to handle the fact that it is clickable/disabled even with user-provided template.

**Comment by maxokorokov** [comment]
Also, @BobbyJoStefflTR, could you please specify "screen readers"? Which onces have you tested with?

**Comment by BobbyJoStefflTR** [comment]
> If I understand correctly the proposal is to replace this `div` by a `button`, but only for the day that is selected I guess:
> 
> https://github.com/ng-bootstrap/ng-bootstrap/blob/f3a7c7d1ffc032952b62eeef4e48f0aa5765c9cf/src/datepicker/datepicker.ts#L129-L135
> 
> Just one comment about the click handling. I'm not sure we'll be able to move it on the `button` element, because the whole contents of the `<div class="ngb-dp-day">...</div>` can be replaced completely by the `[dayTemplate]` input, but we still need to handle the fact that it is clickable/disabled even with user-provided template.

Let me clarify, all of the days would be buttons not just selected days, as they are intended to perform an action of selecting. Unfortunately the click function would need to be on the button in order for this to work for screen readers. Here is W3C's example where they use a button for the date.  [W3C Date Picker Dialog Example](https://www.w3.org/TR/wai-aria-practices-1.1/examples/dialog-modal/datepicker-dialog.html)

I believe this is an issue with any screen reader & browser combination but we tested with Jaws.

**Comment by maxokorokov** [comment]
@BobbyJoStefflTR could you specify the Jaws version please, because they're wildly different

We also might consider replacing this one with the button, maybe:

```html
<div role="gridcell"> <----- this one
    <div class="btn-light ng-star-inserted" ngbdatepickerdayview="">11</div>
</div>
```

Can't predict how it would work out though :)

**Comment by BobbyJoStefflTR** [comment]
> @BobbyJoStefflTR could you specify the Jaws version please, because they're wildly different
> 
> We also might consider replacing this one with the button, maybe:
> 
> ```
> <div role="gridcell"> <----- this one
>     <div class="btn-light ng-star-inserted" ngbdatepickerdayview="">11</div>
> </div>
> ```
> 
> Can't predict how it would work out though :)

Ideally having the `role="gridcell"` is still important, similarly in the example you can see they are using a table and the `td` has an inherit role of `gridcell`. I realize I'm describing the ideal situation and may not always be possible, so just trying to find something that works for all users.

Let me verify exactly what we tested with and get back to you.

**Comment by BobbyJoStefflTR** [comment]
@maxokorokov We [tested these examples](https://ng-bootstrap.github.io/#/components/datepicker/examples) with the following combinations:
**VoiceOver on Mac Catalina:**
* Chrome, most recent version
* Safari, most recent version

**JAWS 2019 on Windows 7:**
* IE11
* Chrome, most recent version
* Firefox, most recent version

**NVDA 2019.1 on Windows 7:**
* IE11
* Chrome, most recent version
* Firefox, most recent version

**Comment by mattdistefano** [comment]
@maxokorokov I could probably PR a fix for this in a couple days if there's interest in moving forward with the proposed solution. I suspect any change here will constitute a breaking change, though.
